src/components/CreateVmWizard/steps/Networking.js

```diff
@@ -0,0 +1,473 @@
+                      title={msg.createVmNetEditActions()}
+                      pullRight
+                    >
+                      <MenuItem
+                        id={`${kebabId}-edit`}
+                        onSelect={() => { this.inlineEditController.onActivate({ rowIndex, rowData }) }}
+                        disabled={actionsDisabled}
+                      >
+                        {msg.edit()}
+                      </MenuItem>
+                      <MenuItem
+                        id={`${kebabId}-delete`}
+                        onSelect={() => { this.onDeleteRow(rowData) }}
+                        disabled={actionsDisabled}
+                      >
+                        {msg.delete()}
+                      </MenuItem>
+                    </DropdownKebab>
+                  </Table.Cell>
+                }
+              </React.Fragment>
+            },
+          ],
+        },
+      },
+    ]
+  }
+
+  onCreateNic () {
+    const newId = generateUnique('NEW_')
+
+    // Search the current NIC names and return the next unused numbered NIC name.
+    let biggestNumber = 0
+    for (const nic of this.props.nics) {
+      const [, number] = /nic(\d+)/.exec(nic.name) || []
+      if (isNumber(number)) {
+        biggestNumber = Math.max(biggestNumber, parseInt(number, 10))
+      }
+    }
+    const nextNicName = `nic${biggestNumber + 1}`
+
+    // If only 1 vnic profile is available, select it automatically
+    const {
+      dataCenterId,
+      cluster,
+      vnicProfiles,
+    } = this.props
+    const vnicList = createVNicProfileList(vnicProfiles, { dataCenterId, cluster })
+    const vnicProfileId =
+      vnicList.length === 1 ? vnicList[0].id : '_'
+
+    // Setup a new nic in the editing object
+    this.setState(state => ({
+      creating: newId,
+      editing: {
+        ...state.editing,
+        [newId]: {
+          id: newId,
+          name: nextNicName,
+          deviceType: NIC_INTERFACE_DEFAULT,
+          vnicProfileId,
+        },
+      },
+    }))
+  }
+
+  onEditNic (rowData) {
+    this.setState(state => ({
+      editing: {
+        ...state.editing,
+        [rowData.id]: rowData,
+      },
+    }))
+  }
+
+  onDeleteRow (rowData) {
+    this.props.onUpdate({ remove: rowData.id })
+  }
+
+  handleCellChange (rowData, field, value) {
+    const editingRow = this.state.editing[rowData.id]
+    if (editingRow) {
+      editingRow[field] = value
+      this.setState(state => ({
+        editing: {
+          ...state.editing,
+          [rowData.id]: editingRow,
+        },
+      }))
+    }
+  }
+
+  // Push the new or editing row up via __onUpdate__
+  handleRowConfirmChange (rowData) {
+    const actionCreate = !!this.state.creating && this.state.creating === rowData.id
+    const editedRow = this.state.editing[rowData.id]
+
+    // TODO: Add field level validation for the edit or create fields
+
+    this.props.onUpdate({ [actionCreate ? 'create' : 'update']: editedRow })
```

Comment:
That will just add clutter.  Either way, using the object literal notation, it would still be `{ [updateType]: editedRow }` which is no less confusing than how it is written now.